service/history/api/updateworkflow/api.go

```diff
@@ -125,9 +129,9 @@ func (u *Updater) Invoke(
 
 func (u *Updater) ApplyRequest(
 	ctx context.Context,
-	workflowLease api.WorkflowLease,
+	updateReg update.Registry,
+	ms workflow.MutableState,
```

Comment:
This interface makes it much clearer what's needed for this function.

Reply:
However, I'd like to see a world in the future where I only need to define a single parameter here.